pkg/utils/netmonitor/netmonitor.go

```diff
@@ -0,0 +1,171 @@
+	Addr  bool
+	Route bool
+}
+
+// InterfacesMonitoring starts the monitoring of the network interfaces.
+// If there is a change in the network interfaces, it will send a message to the channel.
+// With the options, you can choose to monitor only the link, address, or route changes (default: all options are true).
+func InterfacesMonitoring(ctx context.Context, eventChannel chan event.GenericEvent, options *Options) error {
+	// Create channels to receive notifications for link, address, and route changes
+	chLink := make(chan netlink.LinkUpdate)
+	doneLink := make(chan struct{})
+	defer close(doneLink)
+
+	chAddr := make(chan netlink.AddrUpdate)
+	doneAddr := make(chan struct{})
+	defer close(doneAddr)
+
+	chRoute := make(chan netlink.RouteUpdate)
+	doneRoute := make(chan struct{})
+	defer close(doneRoute)
+
+	if options == nil {
+		options = &Options{
+			Link:  true,
+			Addr:  true,
+			Route: true,
+		}
+	}
+
+	// Subscribe to the address updates
+	if err := netlink.AddrSubscribe(chAddr, doneAddr); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	// Subscribe to the link updates
+	if err := netlink.LinkSubscribe(chLink, doneLink); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	// Subscribe to the route updates
+	if err := netlink.RouteSubscribe(chRoute, doneRoute); err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	// Create maps to keep track of interfaces
+	interfaces := make(map[string]bool)
+
+	// Get the list of existing links and add them to the interfaces map
+	links, err := netlink.LinkList()
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	for _, link := range links {
+		interfaces[link.Attrs().Name] = true
+	}
+
+	// Start an infinite loop to handle the notifications
+	for {
+		select {
+		case updateLink := <-chLink:
+			if options.Link {
+				handleLinkUpdate(&updateLink, interfaces, eventChannel)
+			}
+		case updateAddr := <-chAddr:
+			if options.Addr {
+				handleAddrUpdate(&updateAddr, eventChannel)
+			}
+		case updateRoute := <-chRoute:
+			if options.Route {
+				handleRouteUpdate(&updateRoute, eventChannel)
+			}
+		case <-ctx.Done():
+			klog.Info("Stop monitoring network interfaces.")
+			return nil
+		}
+	}
+}
+
+func handleLinkUpdate(updateLink *netlink.LinkUpdate, interfaces map[string]bool, eventChannel chan<- event.GenericEvent) {
+	switch {
+	case updateLink.Header.Type == syscall.RTM_DELLINK:
+		// Link has been removed
+		klog.Info("Interface removed: ", updateLink.Link.Attrs().Name)
+		delete(interfaces, updateLink.Link.Attrs().Name)
+	case !interfaces[updateLink.Link.Attrs().Name] && updateLink.Header.Type == syscall.RTM_NEWLINK:
+		// New link has been added
+		klog.Info("Interface added: ", updateLink.Link.Attrs().Name)
+		interfaces[updateLink.Link.Attrs().Name] = true
+	case updateLink.Header.Type == syscall.RTM_NEWLINK:
+		// Link has been modified
+		if updateLink.Link.Attrs().Flags&net.FlagUp != 0 {
+			klog.Info("Interface ", updateLink.Link.Attrs().Name, " is up")
```

Comment:
```suggestion
			klog.Info("Interface %s is up", updateLink.Link.Attrs().Name)
```

The same for the others